cmd/kube-proxy/app/server_others.go

```diff
@@ -329,6 +336,49 @@ func (s *ProxyServer) setupConntrack() error {
 		}
 	}
 
+	if s.Config.Conntrack.UDPTimeout.Duration > 0 {
+		timeout := int(s.Config.Conntrack.UDPTimeout.Duration / time.Second)
+		if err := ct.SetUDPTimeout(timeout); err != nil {
+			return err
+		}
+	}
+
+	if s.Config.Conntrack.UDPStreamTimeout.Duration > 0 {
+		timeout := int(s.Config.Conntrack.UDPStreamTimeout.Duration / time.Second)
+		if err := ct.SetUDPStreamTimeout(timeout); err != nil {
+			return err
+		}
+	}
+
+	if s.Config.Conntrack.TCPBeLiberal != nil {
+		if *s.Config.Conntrack.TCPBeLiberal {
+			if err := ct.SetTCPBeLiberal(1); err != nil {
+				return err
+			}
+		}
+	} else {
+		// try setting nf_conntrack_tcp_be_liberal=1 if kernelVersion < CTStateInvalidFixKernelVersion
+		// don't return any error, just log them.
+		kernelVersionStr, err := ipvs.NewLinuxKernelHandler().GetKernelVersion()
+		if err != nil {
+			klog.ErrorS(err, "failed to get kernel version")
+			return nil
+		}
+
+		kernelVersion, err := version.ParseGeneric(kernelVersionStr)
+		if err != nil {
+			klog.ErrorS(err, "error parsing kernel version", "kernelVersionStr", kernelVersionStr)
+			return nil
+		}
+
+		if kernelVersion.LessThan(version.MustParseGeneric(CTStateInvalidFixKernelVersion)) {
+			klog.InfoS("SetupConntrack: kernel is not patched with netfilter conntrack INVALID state bug, setting nf_conntrack_tcp_be_liberal=1", "kernelVersion", kernelVersionStr, "fixedKernelVersion", CTStateInvalidFixKernelVersion)
+			if err := ct.SetTCPBeLiberal(1); err != nil {
+				klog.ErrorS(err, "SetupConntrack: failed to set nf_conntrack_tcp_be_liberal=1, conntrack will mark overly delayed packets with out-of-window sequence numbers invalid and kube-proxy will drop them.", "kernelVersion", kernelVersionStr, "fixedKernelVersion", CTStateInvalidFixKernelVersion)
```

Comment:
I'm not sure how much we want to explain here (especially since, as I said, I want to get rid of the drop rule).